### ISSUE TITLE
o.io.quneo QUNEO /pad behaviour

### ISSUE THREAD

**Issue by Pulmunen11** [CNMAT/Music-and-Computing#98]
Hi,

When one has pressed the pads on the QuNeo and then releases, the 'o.io.quneo QUNEO' gives 63 as the value in all x and y pad addresses. For example /pad/01/x --> 63. This might be the default value that the QuNeo itself gives, right? 

This is a problem in that way, that if we map frequency of a resonator to the y-value of the pad, the second we lift finger from the pad, the value returns to 63, which gives annoying glissandos to the resonator sound. It is difficult to try to filter this behaviour out with, say z-values or velocity. I'd be very happy if someone had an easy solution to this problem. 

Thanks!
~ Maija

**Comment by ramagottfried** [comment]
hi maija!
sounds like a continuous vs discrete controller design issue ... 
if you want to "lock" the frequency to a value one you find it on the pad, I'd maybe try using a second button (or maybe a sustain pedal?) to trigger a "save" of that value until you press the store button again... 

but then what do you do if you start playing again after locking the note?  maybe when you release the lock it could slide smoothly to the currently performed value, or something like that?  just thinking out loud, sounds fun: )

cheers!
rama

**Comment by ramagottfried** [comment]
oh!  or, if you use the z-value, you could use a certain z-threshold to lock the pitch, and then you could regain control of the pitch by pressing past that z-threshold again.

**Comment by equilet** [comment]
Hi all - I believe that there is a way to address this more directly in the HW config itself, but I have to check with a quneo when I get home.  Basically, the editor allows for some options for the value in the resting state.  I had a couple of other students ask about this last semester and never remembered to file an issue.  So thank you @Pulmunen11 for bringing this up again!  

Soon,
jml

**Comment by Pulmunen11** [comment]
Hi,

Great to hear from you Rama! Those are interesting suggestions. And I think the students might want to do that in their composition. Like sustaining a sound with pressing another key --> great idea! (And by the way, it is super helpful to have all the material that you created for the 158A. I truly appreciate it very much!)

I think the QuNeo pad x and y going to 63 happens when the pressure (z) gets released to a certain amount or maybe at a certain speed(?), but not when z is 0. It happens before that. So this poses yet another problem. 

Hoping to be able to tackle this problem with the hardware setup. So yeah, Jeff, we'll hopefully have a QuNeo installation party coming up!

Thank you all!

~ Maija

**Comment by equilet** [comment]
OK, I just verified that yes, this return setting can be changed.   Load the QuNeo Editor, click on the various pads, and update the `X Return` and `Y Return` values in the editor.  You can then save this as a preset from there and use it to load into the other QuNeos.  When you click the down arrow of those values past zero, a "latch" option appears.  If you select this option for all pads, the QuNeo then functions the way that @Pulmunen11 was expecting.

**Comment by equilet** [comment]
Worth mentioning that you'll want to save the preset after editing, and then use the "Update preset" option, which updates it on the device itself.

**Comment by equilet** [comment]
@Pulmunen11 @dappertutto - do you want to get together soon to update all of these?  Would it make sense to meet in Wheeler and update them all at once?  I would personally prefer M/Th/F but I cannot do it this particular Friday.

**Comment by EdmundCampion** [comment]
Let's get this into a routine that can be repeated for next offerings of 158A.  I wonder if we can easily write a max patch that accomplishes the same -- i.e. bypass the editor

**Comment by Pulmunen11** [comment]
@equilet I'd be very happy to go to Wheeler to update the QuNeos! So, would Monday still work for you?

And thanks a lot for looking into the problem.

**Comment by equilet** [comment]
Hi again - It looks like walk-in lab hours for Wheeler at 2pm onward on Mondays.  Because of this not lining up with your schedule, I can go with Andrew Harlan to update them if you like?  It's no trouble, and I think this is his preferred schedule as well.

**Comment by equilet** [comment]
Hi @Pulmunen11 

I reprogrammed all of the quneos today w/ Andrew Harlan; we tested them in Wheeler and all seemed fine, but I would love to have someone else's eyes on it just to be sure.

After you have your next lab session:  If the issue is addressed, feel free to close this.

**Comment by Pulmunen11** [comment]
Hi @equilet 

That's great! Thank you so much for doing this.

Most of the QuNeos worked in the right way, but some of them were working in a very weird way. 

Some weird behaviour:
- /x and /y still reseting to 63
- nothing coming in for /velocity 

I tested them all and here are the QuNeos that still are not functioning in correct way:
NODE 
7702
7680
7717
7683
7699
7712
7708

They are in one pile with pink tape around them at 210 Wheeler. My students are making their Project 1 this week and the QuNeos really have to be up and running for them. 

Best,
~ Maija

**Comment by equilet** [comment]
I don't be able to get to it so best to contact @EdmundCampion and @andrewharlan1 
Andrew helped me update the devices and is familiar with the process.  He can work with you to update the remaining ones and test them.

**Comment by equilet** [comment]
"Some weird behaviour:  /x and /y still reseting to 63, nothing coming in for /velocity"

Which pad(s)?

**Comment by equilet** [comment]
@andrewharlan1 it could be the case the the nodes Maija is mentioning still require a firmware update, so you could try that first.

**Comment by Pulmunen11** [comment]
Hi @equilet,

The problems occurred on all pads as far as I know. I was running out of time while testing them, but my estimate is that there were consistently similar problems between all of the 7 devices. 

I have a full week this week with lots of deadlines. We'll just have to manage with those 2 remaining QuNeo's for now and try to fix it next week.

Best,
~ Maija

**Comment by equilet** [comment]
Yes, deadlines on my end too.  Let's adjourn until next week, and in the meantime we'll see what Andrew can help with.

**Comment by equilet** [comment]
Locker code: 5225

**Comment by dappertutto** [comment]
Andrew, I'm afraid I can't come to help you today. If nobody is able to join you, do you think you can do this on your own?

**Comment by Pulmunen11** [comment]
@andrewharlan1 

You're so sweet! Thanks for offering. Would it still be possible for you to join me on Th morning at 9 am? The doors to 210 Wheeler should open then and I have office hours until 9.40. If we can't make them work in that time, you're welcome to stay during the lecture as well. That is, if you have time. I'd be very happy to collaborate on this with you.

Thanks!

~ Maija

**Comment by andrewharlan1** [comment]
Hi Maija and David,

I fixed the 7 additional QuNeo's this afternoon! Not sure what happened with those 7 exactly, but I updated the firmware on all and re-loaded the latch preset, at which point they began working normally. They should all be set, but please let me know if you run into issues again.
